### PR TITLE
Logs: Improve getLogRowContext API

### PR DESCRIPTION
**Pull request by gabor**
give access to the original `query` in `getLogRowContext` . should help with https://github.com/grafana/grafana/pull/52118

### CODE REVIEW

File: public/app/features/explore/LogsContainer.tsx [hunk 1]
```diff
@@ -44,10 +44,11 @@ class LogsContainer extends PureComponent<LogsContainerProps> {
   };
 
   getLogRowContext = async (row: LogRowModel, options?: any): Promise<any> => {
-    const { datasourceInstance } = this.props;
+    const { datasourceInstance, logsQueries } = this.props;
 
     if (hasLogsContextSupport(datasourceInstance)) {
-      return datasourceInstance.getLogRowContext(row, options);
+      const query = (logsQueries ?? []).find((q) => q.refId === row.dataFrame.refId);
```
Comment (gabor) [inline public/app/features/explore/LogsContainer.tsx:50]: i assume there cannot be two queries with the same refId

**Review by ivanahuckova — APPROVED** [review]
LGTM 👍 , but do we need it in `9.0.3`? As this is not a bug fix, I wouldn't be backporting it.

**Comment by gabor** [comment]
> LGTM 👍 , but do we need it in `9.0.3`? As this is not a bug fix, I wouldn't be backporting it.

i agree, but the "reason" for this change, the https://github.com/grafana/grafana/pull/52118 PR is meant to go in in 9.0.3, so it limits our options, i hope it's ok.

**Review by Elfo404 — CHANGES_REQUESTED** [review]
if we modify `DataSourceWithLogsContextSupport` to be 

```
export interface DataSourceWithLogsContextSupport<TQuery extends DataQuery = DataQuery> {
  /**
   * Retrieve context for a given log row
   */
  getLogRowContext: <TContextQueryOptions extends {}>(
    row: LogRowModel,
    options?: TContextQueryOptions,
    query?: TQuery
  ) => Promise<DataQueryResponse>;

  /**
   * This method can be used to show "context" button based on runtime conditions (for example row model data or plugin settings, etc.)
   */
  showContextToggle(row?: LogRowModel): boolean;
}
```
datasources implementing `DataSourceWithLogsContextSupport<LokiQuery>` (here `LokiQuery` is an example) will be limited to have
```
getLogRowContext: (row: LogRowModel, options?: RowContextOptions, query?: LokiQuery) => Promise<{ data: DataFrame[] }>
```
and can rely on Grafana doing the right thing without the need for them to do any runtime check (because in this PR query would be DataQuery).

for completeness, datasources can still implement
```
getLogRowContext: (row: LogRowModel, options?: RowContextOptions, query?: DataQuery) => Promise<{ data: DataFrame[] }>
```
even with my proposal because in `LokiQuery` extends `DataQuery`, so the type is still valid, but it would still make for a nicer API IMHO. wdyt?

**Review by Elfo404 — APPROVED** [review]

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Pull request merged**